api/v1alpha1/clusterextension_types.go

```diff
@@ -73,6 +73,10 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+kubebuilder:Optional
+	// skipCRDUpgradeSafetyCheck specifies whether or not the CRD upgrade safety checks should be skipped when attempting to install the cluster extension.
+	SkipCRDUpgradeSafetyCheck bool `json:"skipCRDUpgradeSafetyCheck,omitempty"`
```

Comment:
What do you think of making a more generic and extensible field here for when we go to add additional preflight checks in the future rather than a field targeted at a single preflight check?

Reply:
I'm thinking maybe something like:
```go
type ClusterExtensionSpec {
...
    Preflight PreflightConfig `json:"preflight,omitempty"`
...
}

type PreflightConfig struct {
    CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig
}

type CRDUpgradeSafetyPreflightConfig struct {
    Skip bool
}
```

With ^ we should be able to add additional fields / functionality to both the CRD Upgrade Safety preflight configuration (like enabling/disabling specific validations) as well as adding entirely new preflight checks to the more general `PreflightConfig` without any breaking changes or adding a bunch of root fields.

Reply:
Alternatively, avoiding a `bool` type, and making it an enum (@joelanford has made this suggestion to me before).

Reply:
Addressed in [latest commit ](https://github.com/operator-framework/operator-controller/pull/845/commits/30bf1c649a88825d181468590011effe5e7ef5a7)